addons/point_of_sale/tests/test_pos_basic_config.py

```diff
@@ -503,3 +503,256 @@ def test_correct_partner_on_invoice_receivables(self):
+                | pos receivable bank |       0 |
+                +---------------------+---------+
+                | Total balance       |     0.0 |
+                +---------------------+---------+
+                """
+        start_qty_available = {
+            self.product1: self.product1.qty_available,
+            self.product2: self.product2.qty_available,
+            self.product3: self.product3.qty_available,
+        }
+
+        self.open_new_session()
+
+        # create orders
+        orders = []
+        orders.append(self.create_ui_order_data(
+            [(self.product1, 3), (self.product2, 2), (self.product3, 1)],
+            payments=[(self.bank_pm, 100)],
+            is_invoiced=True,
+            uid='12346-123-1234',
+            customer=self.customer
+        ))
+        orders.append(self.create_ui_order_data(
+            [(self.product1, -3), (self.product2, -2), (self.product3, -1)],
+            payments=[(self.bank_pm, -100)],
+            uid='12345-123-1234',
+            is_invoiced=True,
+            customer=self.customer
+        ))
+
+        # sync orders
+        order = self.env['pos.order'].create_from_ui(orders)
+
+        # check values before closing the session
+        self.assertEqual(2, self.pos_session.order_count)
+        orders_total = sum(order.amount_total for order in self.pos_session.order_ids)
+        self.assertAlmostEqual(orders_total, self.pos_session.total_payments_amount,
+                               msg='Total order amount should be equal to the total payment amount.')
+        self.assertEqual(
+            self.product1.qty_available,
+            start_qty_available[self.product1],
+        )
+        self.assertEqual(
+            self.product2.qty_available,
+            start_qty_available[self.product2],
+        )
+        self.assertEqual(
+            self.product3.qty_available,
+            start_qty_available[self.product3],
+        )
+
+        # picking and stock moves should be in done state
+        # no exception of return orders
+        for order in self.pos_session.order_ids:
+            self.assertEqual(
+                order.picking_id.state,
+                'done',
+                'Picking should be in done state.'
+            )
+            move_lines = order.picking_id.move_lines
+            self.assertEqual(
+                move_lines.mapped('state'),
+                ['done'] * len(move_lines),
+                'Move Lines should be in done state.'
+            )
+
+        # close the session
+        self.pos_session.action_pos_session_validate()
+
+        # check values after the session is closed
+        session_move = self.pos_session.move_id
+
+        sale_lines = session_move.line_ids.filtered(
+            lambda line: line.account_id == self.sale_account)
+        self.assertEqual(len(sale_lines), 0,
+                         msg='There should be no lines as all is invoiced.')
```

Comment:
Instead of asserting the sale_lines here, can you assert the receivable lines? Maybe check the one with `partner = customer` that its balance is `-100`, while the other receivable line (partner = other customer) should have balance = `100`.